plugins/urls/url.py

```diff
@@ -0,0 +1,43 @@
+import idna
+from kitchen.text.converters import to_unicode
+
+__author__ = 'Gareth Coles'
+
+
+class URL(object):
+    plugin = None
+
+    protocol = None
+    auth = None
+    domain = None
+    port = None
+    path = None
+
+    @property
+    def text(self):
```

Comment:
Yeah, I know that much, but I don't like doing that when things need to be reconstructed, it feels like something that should be static.

Then again, I guess that's what repr is for.